x/ccv/provider/keeper/proposal.go

```diff
@@ -347,30 +324,46 @@ func (k Keeper) BeginBlockInit(ctx sdk.Context) {
 // A prop is included in the returned list if its proposed spawn time has passed.
 //
 // Note: this method is split out from BeginBlockInit to be easily unit tested.
+// TODO JEHAN: Stopping iteration here (dealt with by making a direct call to KVStorePrefixIterator)
 func (k Keeper) ConsumerAdditionPropsToExecute(ctx sdk.Context) []types.ConsumerAdditionProposal {
```

Comment:
This used to call an iterator which was stopped early. Instead, it now uses KVStorePrefixIterator directly